includes/payment-methods/class-upe-split-payment-gateway.php

```diff
@@ -160,13 +160,31 @@ public function should_use_stripe_platform_on_checkout_page() {
 	 * @return UPE_Payment_Method|false UPE payment method instance.
 	 */
 	public function get_selected_payment_method( $payment_method_type ) {
-		if ( $payment_method_type !== $this->stripe_id ) {
-			return false;
+		return WC_Payments::get_payment_method_by_id( $payment_method_type );
+	}
+
+	/**
+	 * Set formatted readable payment method title for order,
+	 * using payment method details from accompanying charge.
+	 *
+	 * @param WC_Order   $order WC Order being processed.
+	 * @param string     $payment_method_type Stripe payment method key.
+	 * @param array|bool $payment_method_details Array of payment method details from charge or false.
+	 */
+	public function set_payment_method_title_for_order( $order, $payment_method_type, $payment_method_details ) {
+		$payment_method = $this->get_selected_payment_method( $payment_method_type );
+		if ( ! $payment_method ) {
+			return;
 		}
 
-		return $this->payment_method;
-	}
+		$payment_method_title = $payment_method->get_title( $payment_method_details );
 
+		$payment_gateway = 'card' === $payment_method->get_id() ? self::GATEWAY_ID : self::GATEWAY_ID . '_' . $payment_method_title;
```

Comment:
```suggestion
		$payment_gateway = $this->id;
```
I think you can safely trust the `id` parameter at this stage of the execution. However, when I thought a little more about this, since this is the only difference between this function and the function it overrides in `UPE_Payment_Gateway`, why don't you just change this line over [here](https://github.com/Automattic/woocommerce-payments/blob/fix/5854-use-correct-payment-method-id/includes/payment-methods/class-upe-payment-gateway.php#L895) to use `$this->id` instead of `self::GATEWAY_ID`? Then we wouldn't need to override this function at all.